spyder/plugins/projects/tests/test_plugin.py

```diff
@@ -289,5 +292,95 @@ def test_project_explorer_tree_root(projects, tmpdir, qtbot):
         assert topleft_index.data() == osp.basename(ppath)
 
 
+@pytest.mark.skipif(sys.platform == 'darwin', reason='Fails in macOS')
+def test_filesystem_notifications(qtbot, projects, tmpdir):
+    """
+    Test that filesystem notifications are emitted when creating,
+    deleting and moving files and directories.
+    """
+    # Create a directory for the project and some files.
+    project_root = tmpdir.mkdir('project0')
+    folder0 = project_root.mkdir('folder0')
+    folder1 = project_root.mkdir('folder1')
+    file0 = project_root.join('file0')
+    file1 = folder0.join('file1')
+    file2 = folder0.join('file2')
+    file3 = folder1.join('file3')
+    file0.write('')
+    file1.write('')
+    file3.write('ab')
+
+    # Open the project
+    projects.open_project(path=to_text_string(project_root))
+
+    # Get a reference to the filesystem event handler
+    fs_handler = projects.watcher.event_handler
+
+    # Test file creation
+    with qtbot.waitSignal(fs_handler.sig_file_created,
+                          timeout=30000) as blocker:
+        file2.write('')
+
+    file_created, is_dir = blocker.args
+    assert file_created == to_text_string(file2)
+    assert not is_dir
+
+    # Test folder creation
+    with qtbot.waitSignal(fs_handler.sig_file_created,
+                          timeout=3000) as blocker:
+        folder2 = project_root.mkdir('folder2')
+
+    folder_created, is_dir = blocker.args
+    assert folder_created == osp.join(to_text_string(project_root), 'folder2')
+
+    # Test file move/renaming
+    new_file = osp.join(to_text_string(folder0), 'new_file')
```

Comment:
I could only make this work if `new_file` is in the same directory as the original file. Is this perhaps a limitation of watchdog?

Reply:
At least on my local setup, it is working as expected, watchdog should be monitoring all the files and folders under the same root folder. I.e., recursive option is set to True

Reply:
Ok, that's great to know!